src/sql/workbench/contrib/commandLine/electron-sandbox/commandLine.ts

```diff
@@ -143,11 +136,18 @@ export class CommandLineWorkbenchContribution implements IWorkbenchContribution,
 				this.logService.warn('Failed to connect due to error: ' + getErrorMessage(err));
 			}
 		}
+
 		if (commandName) {
 			if (this._notificationService) {
 				this._notificationService.status(localize('runningCommandLabel', "Running command: {0}", commandName), { hideAfter: 2500 });
 			}
-			await this._commandService.executeCommand(commandName, connectedContext);
+			if (commandName === Command.connect || commandName === Command.openConnectionDialog) {
+				// Run handlers for 'connect' and 'openConnectionDialog' commands.
+				await this.runCommandHandler(commandName, args);
+			} else {
+				// Execute other commands via commandService.
+				await this._commandService.executeCommand(commandName, connectedContext);
```

Comment:
Make sure to update https://github.com/Microsoft/azuredatastudio/blob/main/src/vs/platform/environment/common/argv.ts#L145, https://github.com/Microsoft/azuredatastudio/blob/main/src/vs/platform/environment/node/argv.ts#L185 and the [docs](https://learn.microsoft.com/en-us/azure-data-studio/command-line#supported-commands) as well

(and if you haven't already it'd be good to go through and make sure the other help options are correct for the various fields)